Essentials/src/com/earth2me/essentials/commands/Commandtogglejail.java

```diff
@@ -101,14 +102,19 @@ public void run(final Server server, final CommandSource sender, final String co
                 player.setJailTimeout(0);
                 player.sendMessage(tl("jailReleasedPlayerNotify"));
                 player.setJail(null);
-                if (player.getBase().isOnline()) {
+                if (player.getBase().isOnline() && ess.getSettings().isTeleportBackWhenFreedFromJail()) {
                     CompletableFuture<Boolean> future = getNewExceptionFuture(sender, commandLabel);
                     player.getAsyncTeleport().back(future);
                     future.thenAccept(success -> {
                         if (success) {
                             sender.sendMessage(tl("jailReleased", player.getName()));
                         }
                     });
+                    future.exceptionally(e -> {
```

Comment:
Not better to call this after `thenAccept`?

```suggestion
future.thenAccept(success -> {
                        if (success) {
                            sender.sendMessage(tl("jailReleased", player.getName()));
                        }
                    }).exceptionally(e -> {
```

Reply:
`thenAccept`/`exceptionally` both return Voided CompletionStages which cannot be chained